README.md

```diff
@@ -31,6 +43,33 @@ source .venv/bin/activate
 pip install -e .
 ```
 
+### Parser structure
+Parsers for each source file format should be in their own file. The parser itself is object-oriented, and the class name should take the format `<Format>Parser` (e.g. `ArxivParser`). XML parsers should inherit `BaseBeautifulSoupParser`. All parser classes should ultimately inherit `IngestBase` (`BaseBeautifulSoupParser` inherits this base class). All parsers will parse the source file into an intermediate dictionary format, outlined in `IngestBase.format`. The intermediate dictionary format must then be converted to the final `ingest_data_model` format via 
+
+```output = self.format(<intermediate_dictionary>, format=...)```
+
+This final formatting into the `ingest_data_model` format is done separately from the parsing in order to simplify any updates that may need to happen if/when the upstream `ingest_data_model` is changed.
+
```

Comment:
Publisher-specific parsers are discouraged unless the publisher uses a proprietary data format.   If a proprietary format is based on a generic format (e.g. DublinCore / OAI-PMH), then the generic parser should be written first, with a publisher-specific parser inheriting from the generic class.